extensions/qute/deployment/src/test/java/io/quarkus/qute/deployment/templatelocator/CustomTemplateLocatorTest.java

```diff
@@ -0,0 +1,236 @@
+import javax.inject.Singleton;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.qute.CheckedTemplate;
+import io.quarkus.qute.EngineBuilder;
+import io.quarkus.qute.Locate;
+import io.quarkus.qute.Location;
+import io.quarkus.qute.Template;
+import io.quarkus.qute.TemplateInstance;
+import io.quarkus.qute.TemplateLocator;
+import io.quarkus.qute.Variant;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class CustomTemplateLocatorTest {
+
+    private static final String TEMPLATE_LOCATION_1 = "/path/to/my/custom/template/basic.html";
+    private static final String TEMPLATE_LOCATION_2 = "/second/path/to/my/custom/template/regular.html";
+    private static final String TEMPLATE_LOCATION_3 = "/third/path/to/my/custom/template/custom.html";
+    private static final String TEMPLATE_LOCATION_4 = "/fourth/path/to/my/custom/template/custom.html";
+    private static final String TEMPLATE_LOCATION_5 = "/fifth/path/to/my/custom/template/custom.html";
+    private static final String CHECKED_TEMPLATE_LOCATION = "my_checked_template_base/myCheckedTemplate";
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(CustomLocator1.class, CustomLocator2.class, CustomLocator3.class, CustomLocatorRegister.class,
+                            TemplateValueProvider.class));
+
+    @Location(TEMPLATE_LOCATION_1)
+    Template template1;
+
+    @Location(TEMPLATE_LOCATION_2)
+    Template template2;
+
+    @Location(TEMPLATE_LOCATION_3)
+    Template template3;
+
+    @Location(TEMPLATE_LOCATION_4)
+    Template template4;
+
+    @Location(TEMPLATE_LOCATION_5)
+    Template template5;
+
+    @Test
+    public void testCustomLocatorRegistration1() {
+        assertEquals("Basic Qute Template!", template1.data("name", "Qute Template").render());
+    }
+
+    @Test
+    public void testCustomLocatorRegistration2() {
+        assertEquals("Regular test template data!", template2.data("name", "test template data").render());
+    }
+
+    @Test
+    public void testCustomLocatorRegistration3() {
+        assertEquals("Custom name!", template3.data("name", "name").render());
+    }
+
+    @Test
+    public void testCustomLocatorRegistration4() {
+        assertEquals("Custom template 4 customName!", template4.data("name", "customName").render());
+    }
+
+    @Test
+    public void testCustomLocatorRegistration5() {
+        assertEquals("Custom template 5 customName!", template5.data("name", "customName").render());
+    }
+
+    @Test
+    public void testIntegerIsIterable() {
+        assertEquals("My Checked Template Number: 3", Templates.myCheckedTemplate(3).render());
+    }
+
+    public static class CustomLocatorRegister {
+
+        @Inject
+        CustomLocator1 customLocator1;
+
+        void addLocator(@Observes EngineBuilder engineBuilder) {
+            engineBuilder
```

Comment:
Would it make sense to register locators annotated with `Locate` automatically?

Reply:
I think so.